fn_test.go

```diff
@@ -952,6 +973,9 @@ func TestRunFunction(t *testing.T) {
 
 	for name, tc := range cases {
 		t.Run(name, func(t *testing.T) {
+			// NOTE: This means we can't run tests in parallel.
+			defaultSource = tc.args.defaultSource
```

Comment:
is there an alternative approach that doesn't rely on global variables?

Reply:
Not sure why I didn't do this initially, but we can read the envvar when we construct the `Function` and store it in a field. This is tidier than the global variable for sure. Updated to do things that way.